hack/k-license/k-license.go

```diff
@@ -0,0 +1,151 @@
+)
+
+func main() {
+	flag.Parse()
+	if *helpFlag {
+		printHelp(os.Args[0])
+		return
+	}
+	excludeList := strings.Split(*excludeDirs, ",")
+	err := filepath.Walk(*path, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if info.IsDir() && contains(excludeList, info.Name()) {
+			return filepath.SkipDir
+		}
+		if !info.IsDir() && (isCodeFile(path) || isBuildFile(path)) && !hasLicense(path) {
+			currentYear := strconv.Itoa(time.Now().Year())
+			err := addLicense(path, *templatesDir, currentYear)
+			if err != nil {
+				return err
+			}
+			fmt.Printf("Modified %s file\n", path)
+		}
+
+		return nil
+	})
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+}
+
+func contains(list []string, str string) bool {
+	for _, item := range list {
+		if item == str {
+			return true
+		}
+	}
+	return false
+}
+
+func printHelp(progName string) {
+	fmt.Fprintf(os.Stderr, "Usage: %s [--help] [-d dir] [--exclude ] [--templates]\n", progName)
+	fmt.Fprintln(os.Stderr, "Add a copyright/license header to all source files in a Git repository.\n-")
+	fmt.Fprintln(os.Stderr, "Options:")
+	flag.PrintDefaults()
+	fmt.Fprintln(os.Stderr, "\nExamples:")
+	fmt.Fprintf(os.Stderr, "  %s               # Add license headers to all source files in the current directory and its subdirectories, using the default file extensions and templates directory.\n", progName)
+	fmt.Fprintf(os.Stderr, "  %s --dir /path/to/repo --exclude vendor,node_modules --templates /path/to/templates   # Add license headers to all .go, .sh, .py, makefile, docker files in /path/to/repo and its subdirectories, using the license header templates in /path/to/templates.\n", progName)
+}
+
+func isCodeFile(path string) bool {
+	ext := strings.ToLower(filepath.Ext(path))
+	switch ext {
+	case ".go", ".c", ".h", ".cpp", ".py", ".ipynb", ".java":
+		return true
+	}
+	return false
+}
+func isBuildFile(path string) bool {
+	switch {
+	case strings.HasSuffix(path, ".sh"):
+		return true
+	case strings.HasSuffix(path, "Makefile"):
+		return true
+	case strings.HasSuffix(path, "Dockerfile"):
+		return true
+	}
+	return false
+}
+
+func hasLicense(path string) bool {
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		return false
+	}
+	return strings.Contains(string(data), "Copyright")
+}
+func getFileType(path string) string {
+	var tmplFilename string
+	switch {
+	case strings.HasSuffix(path, ".sh"):
+		tmplFilename = "boilerplate.sh.txt"
+	case strings.HasSuffix(path, "Makefile"):
+		tmplFilename = "boilerplate.Makefile.txt"
+	case strings.HasSuffix(path, "Dockerfile"):
+		tmplFilename = "boilerplate.Dockerfile.txt"
+	case filepath.Ext(path) == ".py":
+		tmplFilename = "boilerplate.py.txt"
+	case filepath.Ext(path) == ".go":
+		tmplFilename = "boilerplate.go.txt"
+	default:
+		tmplFilename = "boilerplate.tf.txt"
+	}
+	return tmplFilename
+}
+
+func addLicense(path, templatesDir, year string) error {
```

Comment:
Are we handling the exception for generated files - like generated go files? 

Ref: https://github.com/kubernetes/org/blob/main/hack/boilerplate/verify_boilerplate.py#L76-L93

Reply:
Not yet , Thanks for the review i'll get it added